aips/aip-50.md

```diff
@@ -0,0 +1,54 @@
+---
+aip: 50
+title: Allow direct commission change vesting contract
+author: junkil-park, movekevin, wintertoro
+discussions-to: https://github.com/aptos-foundation/AIPs/issues/249
+Status: In Review
+type: Standard Framework
+created: 09/28/2023
+---
+
+# AIP-50 - Change commission rates in delegation pools
+
+## Summary
+Currently the commission rate is set by the pool owner at initiation of pool. We propose allowing the pool owner to change the commission rate after delegation pool creation.
+
+### Goals
+This feature will increase flexibility of pool owners over the network / commission rates. Pool owner can adjust operator commission rates as appropriate, based on market conditions and commercial discussions. As such, this will facilitate staking pools to participate in market dynamics. I.e. the pool can be set at a lower commission to attract more stakers or raise commission when needed. Operators can call the function themselves if they are also the pool owner, or if not, work with the pool owner they are operating for.
+
+### Out of Scope
+Operators will not be able to change commission rate. In our existing design setup, Operators are designated by pool owners.
+
+### Impact
+Delegation pool owners and corresponding operators.
+
+## Alternative solutions
+There is no alternative solution because the delegation pool module currently does not allow chainging the operator commission rate.
+
+Moreover, we decided against allowing operators to change commission rate, as it would not make sense in the context of our existing design setup. Right now, Operators are designated by pool owners so eventually pool owners control everything.
+
+## Reference Implementation
+Please refer to https://github.com/aptos-labs/aptos-core/pull/10226.
```

Comment:
Can we please also list out the high level implementation idea here for easy review? e.g. you mentioned that we will be enforcing a max commission cap, what's the value of that?

Reply:
Yes, I added the "specification" section where I explained what needs be implemented.